x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/queries/TextExpansionQueryBuilder.java

```diff
@@ -223,20 +223,20 @@ protected QueryBuilder doRewrite(QueryRewriteContext queryRewriteContext) {
 
     private QueryBuilder weightedTokensToQuery(String fieldName, TextExpansionResults textExpansionResults) {
         if (tokenPruningConfig != null) {
-            WeightedTokensQueryBuilder weightedTokensQueryBuilder = new WeightedTokensQueryBuilder(
```

Comment:
IMO, weighted tokens was a stop gap that needs to go away. We have it currently because of the old `TextExpansionQueryBuilder`, but I don't think users should ever supply a separate query other than a SparseVectorQueryBuilder, where they can supply their own inference results via `query_vector` or supply a modelText & Id.

Reply:
We're actively using this in Rally so we can't just get rid of it until there's a viable alternative that's live. I intend to keep it for now and remove it at a later date after we can start replacing those use cases with this API call.

Reply:
> We're actively using this in Rally so we can't just get rid of it until there's a viable alternative that's live.

No doubt, but rally should switch to use `SparseVectorQueryBuilder` for all its queries as soon as possible, and then we should deprecate this query.

Reply:
Absolutely, we can add an issue to deprecate it as a followup.